File: examples/java/client/com/example/apiblueprint/runtime/transports/http/HttpApiClient.java
package com.example.apiblueprint.runtime.transports.http;

import com.example.apiblueprint.runtime.runtime.ApiClient;

public class HttpApiClient extends ApiClient {
    public static HttpApiClient create(String baseUrl) {
        return new HttpApiClient(new GenHttpApiConfig(baseUrl));
    }

    public HttpApiClient() {
        this(new GenHttpApiConfig());
    }

    public HttpApiClient(GenHttpApiConfig config) {
        super(new GenJdkHttpApiTransport(config));
    }
}

File: examples/kotlin/client/com/example/apiblueprint/runtime/transports/http/HttpApiClient.kt
package com.example.apiblueprint.runtime.transports.http

import com.example.apiblueprint.runtime.runtime.ApiClient
import okhttp3.OkHttpClient

public fun createHttpApiClient(
    config: HttpApiConfig = HttpApiConfig(),
    httpClient: OkHttpClient = OkHttpClient(),
): ApiClient {
    return ApiClient(OkHttpApiTransport(config, httpClient))
}
